### PR TITLE
Prepare for 0.2.0 release

### LINKED ISSUE #652: Release SBO 0.2.0
1. Increase version to 0.2.0 and release it, but on a new channel, to avoid problems on already existing cluster that consume/work with the old API. 
1. Ensure that bundles created in https://github.com/redhat-developer/service-binding-operator-manifests are versioned from `0.2.0-<commit_number>`
1. Ensure that Travis job that publishes to the Quay is updated
1. Ensure that nightly CI job is updated

Internal tracker: https://issues.redhat.com/browse/APPSVC-663

### PR DESCRIPTION
**Pull request by baijum**
### Motivation

Since 0.2.0 is going to be in another channel. It requires some script changes.

### Changes

Update validation scripts and document release process.

### Testing

Need manual verification of each step.

For further more details refer the CONTRIBUTING.md

 Fixes #652

## TODO

- [x] Prepare release notes template
- [x] Prepare release announement mail template

### CODE REVIEW

File: docs/community-release.md
Comment (DhritiShikhar) [inline docs/community-release.md:1]: nice!

File: Makefile
Comment (pedjak) [inline Makefile:519]: let's call the channel `beta`.

**Review by pedjak — CHANGES_REQUESTED** [review]

File: Makefile
Comment (baijum) [reply Makefile:519]: Are we going to increment API version also into `beta`? cc. @akashshinde

Comment (sbose78) [reply Makefile:519]: I don't think we are in a state to provide beta level compatibility? 🤔 maybe in a future version ( also a future tech preview version )

Comment (pedjak) [reply Makefile:519]: +1 - channel name and API stability level are two orthogonal things, IMHO.

Comment (baijum) [reply Makefile:519]: So, I can change the channel name as `beta`. Is that correct understanding?

**Comment by baijum** [comment]
/hold

This PR can be merged after #608

File: deploy/olm-catalog/service-binding-operator/service-binding-operator.package.yaml
Comment (pedjak) [inline deploy/olm-catalog/service-binding-operator/service-binding-operator.package.yaml:4]: I think you need to revert this changes.

File: deploy/olm-catalog/service-binding-operator/service-binding-operator.package.yaml
Comment (pedjak) [inline deploy/olm-catalog/service-binding-operator/service-binding-operator.package.yaml:5]: you should change this to `beta`

File: deploy/olm-catalog/service-binding-operator/service-binding-operator.package.yaml
Comment (pedjak) [inline deploy/olm-catalog/service-binding-operator/service-binding-operator.package.yaml:6]: default channel should remain alpha, because we do not want to break existing OCP installations.

**Review by pedjak — CHANGES_REQUESTED** [review]

File: deploy/olm-catalog/service-binding-operator/service-binding-operator.package.yaml
Comment (baijum) [reply deploy/olm-catalog/service-binding-operator/service-binding-operator.package.yaml]: Let me know if these changes are fine. ce8ced142aed6e54659609cf8f25031ef4a4c575

File: deploy/olm-catalog/service-binding-operator/service-binding-operator.package.yaml
Comment (baijum) [reply deploy/olm-catalog/service-binding-operator/service-binding-operator.package.yaml:5]: Done. ce8ced142aed6e54659609cf8f25031ef4a4c575

File: deploy/olm-catalog/service-binding-operator/service-binding-operator.package.yaml
Comment (baijum) [reply deploy/olm-catalog/service-binding-operator/service-binding-operator.package.yaml]: Done. ce8ced142aed6e54659609cf8f25031ef4a4c575

File: docs/community-release.md
Comment (Avni-Sharma) [inline docs/community-release.md:3]: Should we modify PR guidelines as well as part of this PR? Need to mention that we would require acceptance tests with each feature PR.

**Review by pedjak — APPROVED** [review]
please squash

**Pull request merged**
